nomos-services/carnot-consensus/src/lib.rs

```diff
@@ -642,6 +642,10 @@ where
                             committee_membership.on_new_block_received(&original_block)
                         },
                     );
+                    // prune old blocks, older than grand parent
+                    new_state.prune_older_blocks_by_view(
+                        new_state.latest_committed_view().prev().prev(),
+                    );
```

Comment:
nit: technically we can remove anything that is before the latest commited block (no need to get its grandparent)

Reply:
yes, we can even keep the last X blocks if that's helpful